test/rfxcom.spec.js

```diff
@@ -4,164 +4,175 @@ var rfxcom = require('../lib'),
     matchers = require('./matchers'),
     protocols = rfxcom.protocols;
 
-describe("RfxCom", function() {
-    beforeEach(function() {
+describe("RfxCom", function () {
+    beforeEach(function () {
         this.addMatchers({
             toHaveSent: matchers.toHaveSent
         });
     });
 
-    describe("RfxCom class", function() {
-        describe("data event handler", function() {
-            it("should emit a response message when it receives message type 0x02", function(done) {
+    describe("RfxCom class", function () {
+        describe("data event handler", function () {
+            it("should emit a response message when it receives message type 0x02", function (done) {
                 var fakeSerialPort = new FakeSerialPort(),
-                    device = new rfxcom.RfxCom("/dev/ttyUSB0", {
+                        device = new rfxcom.RfxCom("/", {
```

Comment:
The issue I hit here was that when the device didn't exist, node-rfxcom threw an exception which wasn't being caught by my node-RED node code (the node-rfxcom client). This caused node-RED to exit ungracefully with an error. Testing for the existence of the device before trying to open it guards against starting node-RED when I've forgotten to plug in the RFXtrx433 resulting in a crash. Hence I needed a device path in the test code which is always guaranteed to exist.

More recently I have been working on gracefully handling plugging in and unplugging the RFXtrx433. This caused me to delve deeper into the problem, and I think I have a better solution now. The uncaught exceptions were being thrown because the default behaviour for Serialport is 'open immediately', however this is done from a process.nextTick() callback so any exception thrown by the open() is on a different call stack and cannot be caught. I'm now manually opening the Serialport only after the event handlers have been installed, so a failing open() emits an error event.

It turns out there is a race condition which occasionally causes the open() to fail even if the device does exist at the time of the test (probably contact bounce when inserting the USB plug) so I have to handle failing open() calls anyway. I will try removing the exit test & see if I still have all the error cases covered. If so, we can happily remove the icky-ness!